docs/csharp/pattern-matching.md

```diff
@@ -205,9 +205,17 @@ Finally, you can add a `null` case to ensure the argument is not `null`:
 
 [!code-csharp[NullCase](../../samples/csharp/PatternMatching/GeometricUtilities.cs#10_NullCase "Add null case")]
 
-The special case for the `null` pattern is interesting because the constant `null` does
-not have a type, but can be converted to any reference type or nullable
-type. 
+The special behavior for the `null` pattern is interesting because the constant
+`null` does not have a type, but can be converted to any reference
+type or nullable type. The value `null` does not match any type pattern,
```

Comment:
I'm not sure why, but I find this sentence "The value `null`..." difficult to understand." and the paragraph too compact and disconnected. Perhaps: 
The special behavior for the `null` pattern is interesting because the constant `null` in the pattern does not have a type but can be converted to any reference type or nullable type. Without this pattern, a `null` value would not match any type pattern, regardless of the compile-time type of the variable. We can see that from the fact that there are no null checks in any of the case blocks of the samples above: they are not necessary, since matching the type pattern guarantees a non-null value.